### PR TITLE
Feat/return params

### PR DESCRIPTION
**Pull request by jeswr**
Branches off #45 

Adds an options parameter to `matchPatternMappings` with the following arguments
 *    skipVar - don't add variables in the quad to the mapping
 *    returnMappings - return the mappings if it is a valid match

### CODE REVIEW

File: lib/QuadTermUtil.ts [hunk 1]
```diff
@@ -307,22 +307,32 @@ export function matchPatternComplete(quad: RDF.BaseQuad, pattern: RDF.BaseQuad):
...
  * @param pattern A pattern - possibly containing variables
+ * @param options
+ *    skipVar - don't add variables in the quad to the mapping
+ *    returnMappings - return the mappings if it is a valid match
  */
-export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad): boolean {
-  const mapping: Record<string, RDF.Term> = {};
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad): boolean;
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings?: false }): boolean;
+export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings: true }): false | Record<string, RDF.Term>;
```
Comment (rubensworks) [inline lib/QuadTermUtil.ts]: Shouldn't this become the following?

```suggestion
export function matchPatternMappings(quad: RDF.BaseQuad, pattern: RDF.BaseQuad, opt: { skipVar?: boolean; returnMappings: true }): Record<string, RDF.Term>;
```

File: lib/QuadTermUtil.ts [hunk 2]
```diff
@@ -307,22 +307,32 @@ export function matchPatternComplete(quad: RDF.BaseQuad, pattern: RDF.BaseQuad):
  *
  * @param quad A quad - possibly containing variables
  * @param pattern A pattern - possibly containing variables
+ * @param options
+ *    skipVar - don't add variables in the quad to the mapping
```
Comment (rubensworks) [inline lib/QuadTermUtil.ts]: What does this mean exactly?
Will variables in the quad match with anything from the pattern?

**Review by rubensworks — CHANGES_REQUESTED** [review]
Looks useful!

Could you also document this functionality in the README?

File: lib/QuadTermUtil.ts (hunk 1 above)
Comment (jeswr) [reply lib/QuadTermUtil.ts]: No - as it returns false if there isn't a valid mapping (e.g. trying to map `ruben a person` to `?v ?v ?v`, we don't know if `?v` should be `ruben`, `a` or `person`)

File: lib/QuadTermUtil.ts (hunk 2 above)
Comment (jeswr) [reply lib/QuadTermUtil.ts]: Let quad be `?s a ?s` and pattern be `?a ?b ?c`.

If `skipVar` is set to true, then this match is fine. If `skipVar` is set to false then there is a conflict between trying to set ?s = ?a and ?s = ?c so it returns false

File: lib/QuadTermUtil.ts (hunk 1 above)
Comment (rubensworks) [reply lib/QuadTermUtil.ts]: Ok, that makes sense.

File: lib/QuadTermUtil.ts (hunk 2 above)
Comment (rubensworks) [reply lib/QuadTermUtil.ts]: Ok, that's clear. We'll need a more descriptive comment to explain this behaviour though (and maybe also a different option name, but can't think of any suggestions atm).

Comment (jeswr) [reply lib/QuadTermUtil.ts]: Perhaps mapVar, where mapVar = !skipVar?

Comment (rubensworks) [reply lib/QuadTermUtil.ts]: Not sure. Maybe something in the direction of `skipVarIdentity`?

Comment (jeswr) [reply lib/QuadTermUtil.ts]: Not sure - `skipVarMapping`?

Comment (jeswr) [reply lib/QuadTermUtil.ts]: or `skipMappingVar`?

Comment (rubensworks) [reply lib/QuadTermUtil.ts]: Ok, let's go with `skipVarMapping`.

**Review by rubensworks — APPROVED** [review]
Looks good!

Could you also document this functionality in the README?

**Comment by jeswr** [comment]
Done @rubensworks - sorry for the delay

**Comment by rubensworks** [comment]
Thanks! Released as 1.9.0.

**Pull request merged**